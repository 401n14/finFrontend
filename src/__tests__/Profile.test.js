import React from 'react';
import Profile from '../views/Profile';
import {configure, mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { useAuth0 } from '../react-auth0-spa.js';


configure({adapter:new Adapter()});

const user = {
    email: 'johndoe@me.com',
    email_verified: true,
    sub: 'google-oauth2|2147627834623744883746',
  };

  jest.mock('../react-auth0-spa.js');



describe('Snapshot Test', () => {
    beforeEach(() => {
        // Mock the Auth0 hook and make it return a logged in state
        useAuth0.mockReturnValue({
            isAuthenticated: true,
            user,
            logout: jest.fn(),
            loginWithRedirect: jest.fn(),
        });
    });
    it('renders correctly, Snapshot', () => {
        const page = renderer.create(<Profile />).toJSON();
        expect(page).toMatchSnapshot();
    });
});
describe('Renders without a user', () => {
    beforeEach(() => {
        // Mock the Auth0 hook and make it return a logged in state
        useAuth0.mockReturnValue({
            isAuthenticated: true,
            logout: jest.fn(),
            loginWithRedirect: jest.fn(),
            loading: jest.fn(),
        });
    });
    it('renders loading page', () => {
        let wrapper = shallow(<Profile />)
        expect(wrapper.text()).toEqual('<Loading />');

    });
});