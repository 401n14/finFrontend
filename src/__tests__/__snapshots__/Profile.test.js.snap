// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test renders correctly, Snapshot 1`] = `
<div
  className="mb-5 container"
>
  <div
    className="align-items-center profile-header mb-5 text-center text-md-left row"
  >
    <div
      className="col-md-2"
    >
      <img
        alt="Profile"
        className="rounded-circle img-fluid profile-picture mb-3 mb-md-0"
      />
    </div>
    <div
      className="col-md"
    >
      <h2 />
      <p
        className="lead text-muted"
      >
        johndoe@me.com
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <pre
      className="rounded"
    >
      <code
        className="json"
      >
        {
  "email": "johndoe@me.com",
  "email_verified": true,
  "sub": "google-oauth2|2147627834623744883746"
}
      </code>
    </pre>
  </div>
</div>
`;
